api/v1alpha1/jwt_types.go

```diff
@@ -81,3 +85,28 @@ type ClaimToHeader struct {
 	// to separate the JSON name path.
 	Claim string `json:"claim"`
 }
+
+// JWTHeaderExtractor defines an HTTP header location to extract JWT token
+type JWTHeaderExtractor struct {
+	// Name is the HTTP header name to retrieve the token
+	//
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// ValuePrefix is the prefix that should be stripped before extracting the token.
+	// The format would be used by Envoy like "{ValuePrefix}<TOKEN>".
+	// For example, "Authorization: Bearer <TOKEN>", then the ValuePrefix="Bearer " with a space at the end.
+	ValuePrefix *string `json:"valuePrefix,omitempty"`
+}
+
+// JWTExtractor defines a custom JWT token extraction from HTTP request.
+type JWTExtractor struct {
+	// Headers represents a list of HTTP request headers to extract the JWT token from.
+	Headers []JWTHeaderExtractor `json:"headers,omitempty"`
+
+	// Cookies represents a list of cookie names to extract the JWT token from.
+	Cookies []string `json:"cookies,omitempty"`
+
+	// Params represents a list of query parameters to extract the JWT token from.
+	Params []string `json:"params,omitempty"`
```

Comment:
is there a strong case for `Params` & `Headers` ? should be add it later and only add `Cookies` for now ?

Reply:
IMO, it is just yet another option to specify the desired location to extract JWT, omitting `Params` or `Headers` might not be necessary for my use case yet. But considering that those attributes are also ways to extract JWT tokens from different use cases, I included them now. 

But `add it later` is also works for me, thought @arkodg?

Reply:
yeah lets add it later, prefer use case driven development

Reply:
noted @arkodg